oauth2/handler.go

```diff
@@ -975,9 +976,13 @@ func (h *Handler) oauth2TokenExchange(w http.ResponseWriter, r *http.Request) {
 		}
 
 		// only for client_credentials, otherwise Authentication is included in session
-		if accessRequest.GetGrantTypes().ExactOne("client_credentials") {
+		if accessRequest.GetGrantTypes().ExactOne(string(fosite.GrantTypeClientCredentials)) {
 			session.Subject = accessRequest.GetClient().GetID()
 		}
+		// only for password grant, otherwise Authentication is included in session
+		if accessRequest.GetGrantTypes().ExactOne(string(fosite.GrantTypePassword)) {
+			session.Subject = accessRequest.GetRequestForm().Get("username")
```

Comment:
In the auth code flow, we write the identity ID to the sub claim. Shouldn't this behave the same way?

Reply:
And put the username under `ext` then?

Reply:
Hm, I guess that's true. Where does "username" come from here? In Kratos we mainly deal with emails, right?

Reply:
I tend to agree with Jonas. Isn't there a standard `username` claim?

Reply:
We might have to clarify with the customer.

Reply:
Username is not a standard claim.

Reply:
`username` in the code above is coming from the token request params (https://datatracker.ietf.org/doc/html/rfc6749#section-4.3.2)